### ISSUE TITLE
Keep track on the building warning

### ISSUE THREAD

**Issue by xytintel** [intel/torch-xpu-ops#1594]
We need to continuously check for building warning:

- [x] /home/sdp/intel/oneapi/compiler/2025.0/bin/compiler/../../include/sycl/detail/builtins/builtins.hpp:235:1: warning: multi-line comment [-Wcomment] https://github.com/intel/llvm/issues/15063
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/MultiTensorApply.h:170:34: warning: 'HostAlloc' is deprecated: at::xpu::HostAlloc(...) is deprecated. Please use at::getHostAllocator(at::kXPU)->allocate(...) instead. [-Wdeprecated-declarations]
  170 |   auto tlAddress_dptr = at::xpu::HostAlloc(
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/MultiTensorApply.h:191:12: warning: 'CachingHostAllocator_recordEvent' is deprecated: at::xpu::CachingHostAllocator_recordEvent(...) is deprecated. Please use at::getHostAllocator(at::kXPU)->record_event(...) instead. [-Wdeprecated-declarations]
  191 |   at::xpu::CachingHostAllocator_recordEvent(
 - [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/Dequant_int4.cpp:33:9: warning: unused variable ‘nsg_n’ [-Wunused-variable]
   33 |     int nsg_n = n / GroupN;
- [ ] /home/sdp/intel/oneapi/compiler/2025.0/bin/compiler/../../include/sycl/accessor.hpp:2588:5: warning: ‘throw’ will always call ‘terminate’ [-Wterminate]
 2588 |     throw sycl::exception(
      |     ^~~~~~~~~~~~~~~~~~~~~~
 2589 |         make_error_code(errc::invalid),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2590 |         "get_multi_ptr must not be called on the host for a local accessor");
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/sparse/xpu/sycl/SparseSoftmaxKernels.cpp:399:13: warning: unused variable ‘pool_index’ [-Wunused-variable]
  399 |     int64_t pool_index = 0;
      |             ^~~~~~~~~~
/home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/sparse/xpu/sycl/SparseSoftmaxKernels.cpp:383:8: warning: unused variable ‘strides_ptr’ [-Wunused-variable]
  383 |   auto strides_ptr = strides.data_ptr<int64_t>();
      |        ^~~~~~~~~~~
/home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/sparse/xpu/sycl/SparseSoftmaxKernels.cpp:395:8: warning: variable ‘indices_accessor’ set but not used [-Wunused-but-set-variable]
  395 |   auto indices_accessor = indices.packed_accessor64<int64_t, 2>();
      |        ^~~~~~~~~~~~~~~~
- [ ] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/pstl/PSTLFunctions.h:360:41: warning: comparison of unsigned expression in ‘>= 0’ is always true [-Wtype-limits]
  360 |       for (int64_t _k = 1; item_id - _k >= 0; _k++) {
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp:259:51: warning: ‘out_cl_h_stride’ is used uninitialized [-Wuninitialized]
  259 |   MaxPool3dKerenlFunctor<scalar_t, channels_last> kfn(
- [x] [6048/8170] Building SYCL (Device) object torch-xpu-ops-sycl-FractionalMaxPool2dKernels_gen_FractionalMaxPool2dKernels.cpp.o
/home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/FractionalMaxPool2dKernels.cpp:101:7: warning: 'PackedTensorAccessor' is deprecated [-Wdeprecated-declarations]
  101 |       PackedTensorAccessor<int64_t, 4> indices,
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/ScanUtils.h:59:32: warning: variable ‘id’ set but not used [-Wunused-but-set-variable]
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/LerpKernels.cpp:17:14: warning: variable ‘self_val_f’ set but not used [-Wunused-but-set-variable]
   17 |     opmath_t self_val_f = self_val;
      |              ^~~~~~~~~~
/home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/LerpKernels.cpp:18:14: warning: variable ‘end_val_f’ set but not used [-Wunused-but-set-variable]
   18 |     opmath_t end_val_f = end_val;
      |              ^~~~~~~~~
/home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/LerpKernels.cpp:19:14: warning: variable ‘weight_val_f’ set but not used [-Wunused-but-set-variable]
   19 |     opmath_t weight_val_f = weight_val;
- [ ] warning: Adding 4 occurrences of additional control flow due to presence of generic address space operations
in function _ZTSN2at6native3xpu16NMSKernelFunctorIN3c104HalfEfEE (Enable PrintVerboseGenericControlFlowLog flag to acquire detailed log. Requires debuginfo!)
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/BatchNormKernels.cpp:5420:12: warning: 'value_or_else<at::Tensor, (lambda at /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/BatchNormKernels.cpp:5420:43)>' is deprecated: Please use std::optional::value_or instead of c10::value_or_else [-Wdeprecated-declarations]
 5420 |       c10::value_or_else(running_var_opt, [] { return Tensor(); });
      |            ^
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/sparse/xpu/sycl/SparseCsrTensorMathKernels.cpp:251:8: warning: unused variable ‘numel’ [-Wunused-variable]
  251 |   auto numel = values.numel();
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/UpSampleBilinear2dKernels.cpp:422:23: warning: unused variable ‘scale_w’ [-Wunused-variable]
  422 |           accscalar_t scale_w =
      |                       ^~~~~~~
/home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/UpSampleBilinear2dKernels.cpp:424:23: warning: unused variable ‘scale_h’ [-Wunused-variable]
  424 |           accscalar_t scale_h =
- [x] /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/Indexing.cpp:1461:23: warning: unused variable ‘defaultMaxGroupThreads’ [-Wunused-variable]
 1461 |                   int defaultMaxGroupThreads = syclMaxWorkGroupSize(caller);

**Comment by xytintel** [comment]
https://github.com/intel/torch-xpu-ops/pull/1595 Resolved warnings:
- /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/MultiTensorApply.h:170:34: warning: 'HostAlloc' is deprecated: at::xpu::HostAlloc(...) is deprecated. Please use at::getHostAllocator(at::kXPU)->allocate(...) instead. [-Wdeprecated-declarations]
170 | auto tlAddress_dptr = at::xpu::HostAlloc(
- /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/MultiTensorApply.h:191:12: warning: 'CachingHostAllocator_recordEvent' is deprecated: at::xpu::CachingHostAllocator_recordEvent(...) is deprecated. Please use at::getHostAllocator(at::kXPU)->record_event(...) instead. [-Wdeprecated-declarations]
191 | at::xpu::CachingHostAllocator_recordEvent(
- /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/Dequant_int4.cpp:33:9: warning: unused variable ‘nsg_n’ [-Wunused-variable]
33 | int nsg_n = n / GroupN;
- /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/sparse/xpu/sycl/SparseSoftmaxKernels.cpp:399:13: warning: unused variable ‘pool_index’ [-Wunused-variable]
399 | int64_t pool_index = 0;
| ^~~~~~~~~~
/home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/sparse/xpu/sycl/SparseSoftmaxKernels.cpp:383:8: warning: unused variable ‘strides_ptr’ [-Wunused-variable]
383 | auto strides_ptr = strides.data_ptr();
| ^~~~~~~~~~~
/home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/sparse/xpu/sycl/SparseSoftmaxKernels.cpp:395:8: warning: variable ‘indices_accessor’ set but not used [-Wunused-but-set-variable]
395 | auto indices_accessor = indices.packed_accessor64();
| ^~~~~~~~~~~~~~~~
- /home/sdp/xyt/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/BatchNormKernels.cpp:5420:12: warning: 'value_or_else' is deprecated: Please use std::optional::value_or instead of c10::value_or_else [-Wdeprecated-declarations]
5420 | c10::value_or_else(running_var_opt, [] { return Tensor(); });
| ^

**Comment by xytintel** [comment]
> [@xytintel](https://github.com/xytintel) please check [#146](https://github.com/intel/torch-xpu-ops/issues/146) whether need consider in this topic also

@chuanqi129 Yes, we will file another PR to take care of it.

**Comment by chuanqi129** [comment]
@xytintel for those unsolved warnings, do we still plan to solve them? If so, can we re-open this issue?

**Comment by EikanWang** [comment]
Let re-open this issue. It does not make sense to close an issue without solving all the items.

**Comment by CuiYifeng** [comment]
Update build warnings: one old bullet left and two new bullets added.

**Comment by CuiYifeng** [comment]
Close this issue since `Werror` is enabled.
